src/test/java/seedu/address/storage/JsonAdaptedOwnerTest.java

```diff
@@ -33,54 +36,71 @@ public void toModelType_validOwnerDetails_returnsPerson() throws Exception {
         assertThrows(IllegalValueException.class, expectedMessage, owner::toModelType);
     }
 
     @Test
     public void toModelType_invalidPhone_throwsIllegalValueException() {
         JsonAdaptedOwner owner =
-                new JsonAdaptedOwner(VALID_NAME, INVALID_PHONE, VALID_EMAIL, VALID_ADDRESS);
+                new JsonAdaptedOwner(VALID_IC_NUMBER, VALID_NAME, INVALID_PHONE, VALID_EMAIL, VALID_ADDRESS);
         String expectedMessage = Phone.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, owner::toModelType);
     }
 
     @Test
     public void toModelType_invalidEmail_throwsIllegalValueException() {
         JsonAdaptedOwner owner =
-                new JsonAdaptedOwner(VALID_NAME, VALID_PHONE, INVALID_EMAIL, VALID_ADDRESS);
+                new JsonAdaptedOwner(VALID_IC_NUMBER, VALID_NAME, VALID_PHONE, INVALID_EMAIL, VALID_ADDRESS);
         String expectedMessage = Email.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, owner::toModelType);
     }
 
     @Test
     public void toModelType_nullEmail_throwsIllegalValueException() {
-        JsonAdaptedOwner owner = new JsonAdaptedOwner(VALID_NAME, VALID_PHONE, null, VALID_ADDRESS);
+        JsonAdaptedOwner owner = new JsonAdaptedOwner(VALID_IC_NUMBER, VALID_NAME, VALID_PHONE, null, VALID_ADDRESS);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Email.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, owner::toModelType);
     }
 
     @Test
     public void toModelType_invalidAddress_throwsIllegalValueException() {
         JsonAdaptedOwner owner =
-                new JsonAdaptedOwner(VALID_NAME, VALID_PHONE, VALID_EMAIL, INVALID_ADDRESS);
+                new JsonAdaptedOwner(VALID_IC_NUMBER, VALID_NAME, VALID_PHONE, VALID_EMAIL, INVALID_ADDRESS);
         String expectedMessage = Address.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, owner::toModelType);
     }
 
     @Test
     public void toModelType_nullAddress_throwsIllegalValueException() {
-        JsonAdaptedOwner owner = new JsonAdaptedOwner(VALID_NAME, VALID_PHONE, VALID_EMAIL, null);
+        JsonAdaptedOwner owner = new JsonAdaptedOwner(VALID_IC_NUMBER, VALID_NAME, VALID_PHONE, VALID_EMAIL, null);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Address.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, owner::toModelType);
     }
 
+    @Test
+    public void toModelType_invalidIcNumber_throwsIllegalValueException() {
+        JsonAdaptedOwner owner =
+                new JsonAdaptedOwner(INVALID_IC_NUMBER, VALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_ADDRESS);
+        String expectedMessage = IdentificationCardNumber.MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, owner::toModelType);
+    }
+
+    @Test
```

Comment:
Thanks for implementing all these test cases to check the accuracy of the software